### PR TITLE
create, rename, delete container

### PR DESCRIPTION
**Pull request by appukuttan-shailesh**
Implements #24

### CODE REVIEW

File: hbp_archive.py [hunk 1]
```diff
@@ -1035,6 +1041,73 @@ def _get_container_info(self):
...
+        public : boolean, optional
+            specify if container is to be made public; default is private
+
+        Note
+        ----
+        Use restricted to Superusers/Operators.
+        """
+        if container_name in self.container_names:
+            raise Exception("Container named '{}' already exists!".format(container_name))
+        response = self._connection.put_container(container_name)
```
Comment (apdavison) [inline hbp_archive.py]: We should check that the response is successful, and raise an Exception otherwise

File: hbp_archive.py [hunk 2]
```diff
@@ -1035,6 +1041,73 @@ def _get_container_info(self):
...
+        Note
+        ----
+        Use restricted to Superusers/Operators.
+        """
+        raise NotImplementedError("It is not possible to directly rename a container."
+            "\nSee 'https://bugs.launchpad.net/swift/+bug/1231540' for more details."
+            "\nWorkaround: copy contents of existing container to a new container "
+            "(with desired name) and then delete the old container.")
+
+    def delete_container(self, container_name, prompt=True):
```
Comment (apdavison) [inline hbp_archive.py]: I don't think we should make the prompt optional. The user should _always_ be asked to confirm deletion.

File: hbp_archive.py [hunk 3]
```diff
@@ -1035,6 +1041,73 @@ def _get_container_info(self):
...
+            print("Are you sure you wish to delete the container named '{}' containing '{}' item(s)?".format(container_name, c.count()))
+            print("If yes, type in the name of the container to proceed. Any other input will cancel this operation.")
+            c_name = raw_input("Input: ")
+            if c_name != container_name:
+                logger.info("Operation cancelled. Container '{}' is NOT deleted.".format(container_name))
+                return
+        items = c.list()
+        for item in items:
+            c.delete(item.name)
+        response = self._connection.delete_container(container_name)
```
Comment (apdavison) [inline hbp_archive.py]: Again, we should probably check that the response was successful

File: hbp_archive.py (hunk 1 above)
Comment (appukuttan-shailesh) [reply hbp_archive.py]: Actually there is no response obtained (I need to remove that variable... It was placed to check for the same). When it is successful, there is no response. When there is an issue, it throws an error.

Post the main statement, I could check if a container by the specified name now exists. But this is probably useless in view of the above (if no error then its created).

Shall I just remove the `response` variable?

File: hbp_archive.py (hunk 3 above)
Comment (appukuttan-shailesh) [reply hbp_archive.py]: Same as first comment. A check could be done to verify that the container is deleted... but this is anyways indicated by the lack of an error message.

File: hbp_archive.py (hunk 2 above)
Comment (appukuttan-shailesh) [reply hbp_archive.py]: My intention of providing the prompt as an option was to enable writing of automatable scripts which might require the creation of temporary disposable containers. Having a prompt within these would require used intervention. Do you think we should keep it optional in view of this?

Comment (appukuttan-shailesh) [reply hbp_archive.py]: @apdavison ... how do you think we proceed on this?

Comment (apdavison) [reply hbp_archive.py]: My impression is that we should not be using CSCS storage for "temporary disposable containers". Can you give more details on the use case?

File: hbp_archive.py (hunk 1 above)
Comment (appukuttan-shailesh) [reply hbp_archive.py]: @apdavison : your comments on these (and others)?

File: hbp_archive.py (hunk 2 above)
Comment (appukuttan-shailesh) [reply hbp_archive.py]: At the time of implementing this, it was with an eye on possible requirements for the model curation process.... as an intermediate staging area before the models complete curation. Access rights could be restricted to just the curators and the model owners.... But as discussed later on, the staging area might not be necessary at all. If so, then no other use case for the moment. So make the delete prompt mandatory?

Comment (apdavison) [reply hbp_archive.py]: yes, at least for now, I think that's the safest course. We can always revisit this later if the need arises.

File: hbp_archive.py (hunk 1 above)
Comment (apdavison) [reply hbp_archive.py]: ok, if `response` is `None` on success, then you can forget about my comment. Maybe add a comment in the code explaining that `put_container` doesn't return anything on success.

Comment (appukuttan-shailesh) [reply hbp_archive.py]: Done.

File: hbp_archive.py (hunk 2 above)
Comment (appukuttan-shailesh) [reply hbp_archive.py]: Ok.

**Comment by appukuttan-shailesh** [comment]
Changes made.

**Pull request merged**